pyomo/repn/plugins/gams_writer.py

```diff
@@ -326,15 +333,17 @@ def _write_model(self,
                 if v.is_binary():
                     binary.append(var)
                 elif v.is_integer():
-                    if v.bounds == (0,1):
+                    if (v.has_lb() and (value(v.lb) >= 0)) and \
+                       (v.has_ub() and (value(v.ub) <= 1)):
```

Comment:
Mostly for consistently with the is_binary method in core/kernel. That additional manual "is binary" check will always be False once the AML interface is re-implemented on top of core/kernel. It should stay there for now.

As for your real question, I don't know why I went with that other than that if I started with a binary variable and then set its upper bound to 0 (instead of explicitly fixing it), I wouldn't expect to see it categorized as a general integer variable. In core/kernel there is no distinction between binary and integer variables other than by looking at the bounds (anyone is free to debate this though once we start doing a design review in core/kernel).